Sources/SwiftlySearch/SwiftlySearch.swift

```diff
@@ -21,17 +21,29 @@ import SwiftUI
 import Combine
 
 public extension View {
-    func navigationBarSearch(_ searchText: Binding<String>) -> some View {
-        return overlay(SearchBar(text: searchText).frame(width: 0, height: 0))
+    func navigationBarSearch(_ searchText: Binding<String>, placeholder: String? = nil, hidesNavigationBarDuringPresentation: Bool = true, hidesSearchBarWhenScrolling: Bool = true) -> some View {
+        return overlay(SearchBar<AnyView>(text: searchText, placeholder: placeholder, hidesNavigationBarDuringPresentation: hidesNavigationBarDuringPresentation, hidesSearchBarWhenScrolling: hidesSearchBarWhenScrolling).frame(width: 0, height: 0))
+    }
+    
+    func navigationBarSearch<ResultContent: View>(_ searchText: Binding<String>, placeholder: String? = nil, hidesNavigationBarDuringPresentation: Bool = true, hidesSearchBarWhenScrolling: Bool = true, @ViewBuilder resultContent: @escaping (String) -> ResultContent) -> some View {
+        return overlay(SearchBar(text: searchText, placeholder: placeholder, hidesNavigationBarDuringPresentation: hidesNavigationBarDuringPresentation, hidesSearchBarWhenScrolling: hidesSearchBarWhenScrolling, resultContent: resultContent).frame(width: 0, height: 0))
     }
```

Comment:
Is there a way to make these into a single function definition instead of having to rely on overloading?

Reply:
I tried some things, but couldn't find a good solution. It seems like overloading is the best bet for handling optional generics. (e.g. discussed here: [Stackoverflow](https://stackoverflow.com/questions/28458160/default-value-for-optional-generic-parameter-in-swift-function))

Reply:
That's a shame. Ah well. Maybe in a future version of Swift there will be a better solution.